helm/v1beta1/types_helm_chart_repository.go

```diff
@@ -72,17 +71,45 @@ type ConnectionConfig struct {
 	// It is used as a trust anchor to validate the TLS certificate presented by the remote server.
 	// The key "ca-bundle.crt" is used to locate the data.
 	// If empty, the default system roots are used.
-	// The namespace for this config map is openshift-config.
+	// The namespace for this config map is openshift-config for Cluster scope repositoty if not specified by the user.
+	// For ProjectHelmChartRepository the default will be the namespace where the repository is getting instantiated.
 	// +optional
-	CA configv1.ConfigMapNameReference `json:"ca,omitempty"`
+	CA ConfigMapNameReference `json:"ca,omitempty"`
 
 	// tlsClientConfig is an optional reference to a secret by name that contains the
 	// PEM-encoded TLS client certificate and private key to present when connecting to the server.
 	// The key "tls.crt" is used to locate the client certificate.
 	// The key "tls.key" is used to locate the private key.
-	// The namespace for this secret is openshift-config.
+	// The namespace for this secret is openshift-config for Cluster scope repository if not specified by the user.
+	// For ProjectHelmChartRepository the default will be the namespace where the repository is getting instantiated.
 	// +optional
-	TLSClientConfig configv1.SecretNameReference `json:"tlsClientConfig,omitempty"`
+	TLSClientConfig SecretNamespacedReference `json:"tlsClientConfig,omitempty"`
+}
+
+// SecretNamespacedReference references a secret in a specific namespace.
+// The namespace must be specified at the point of use.
+type SecretNamespacedReference struct {
+	// name is the metadata.name of the referenced secret
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+	// namespace is the metadata.namespace of the referenced secret
+	// +kubebuilder:validation:Optional
+	// +optional
```

Comment:
You only need one of these, so we tend to just use `+optional`

Reply:
> You only need one of these, so we tend to just use +optional

Given that for required, we have to use `+kubebuilder:validation:Required`, you'd think we'd prefer to use `+kubebuilder:validation:Optional` for optional items, for symmetry.